taskcat/stacker.py

```diff
@@ -660,6 +675,231 @@ def generate_uuid(self, uuid_type):
+                # Determines if AWS QuickStart default KeyPair name has been requested
+                getkeypair_re = re.compile('\$\[\w+_getkeypair]', re.IGNORECASE)
+
+                # Determines if AWS QuickStart default license bucket name has been requested
+                getlicensebucket_re = re.compile('\$\[\w+_getlicensebucket]', re.IGNORECASE)
+
+                # Determines if AWS QuickStart default media bucket name has been requested
+                getmediabucket_re = re.compile('\$\[\w+_getmediabucket]', re.IGNORECASE)
+
+                # Determines if license content has been requested
+                licensecontent_re = re.compile('\$\[\w+_getlicensecontent]', re.IGNORECASE)
+
+                # Determines if s3 replacement was requested
+                gets3replace = re.compile('\$\[\w+_url_.+]$', re.IGNORECASE)
+                geturl_re = re.compile('(?<=._url_)(.+)(?=]$)', re.IGNORECASE)
+
+                if gen_string_re.search(param_value):
+                    random_string = self.regxfind(gen_string_re, param_value)
+                    param_value = self.generate_random('alpha', 20)
+
+                    if self.verbose:
+                        print("{}Generating random string for {}".format(D, random_string))
+                    parmdict['ParameterValue'] = param_value
+
+                if gen_numbers_re.search(param_value):
+                    random_numbers = self.regxfind(gen_numbers_re, param_value)
+                    param_value = self.generate_random('number', 20)
+
+                    if self.verbose:
+                        print("{}Generating numeric string for {}".format(D, random_numbers))
+                    parmdict['ParameterValue'] = param_value
+
+                if genuuid_re.search(param_value):
+                    uuid_string = self.regxfind(genuuid_re, param_value)
+                    param_value = self.generate_uuid('A')
+
+                    if self.verbose:
+                        print("{}Generating random uuid string for {}".format(D, uuid_string))
+                    parmdict['ParameterValue'] = param_value
+
+                if autobucket_re.search(param_value):
+                    bkt = self.regxfind(autobucket_re, param_value)
+                    param_value = self.get_s3bucket()
+                    if self.verbose:
+                        print("{}Setting value to {}".format(D, bkt))
+                    parmdict['ParameterValue'] = param_value
+
+                if gets3replace.search(param_value):
+                    url = self.regxfind(geturl_re, param_value)
+                    param_value = self.get_s3contents(url)
+                    if self.verbose:
+                        print("{}Raw content of url {}".format(D, url))
+                    parmdict['ParameterValue'] = param_value
+
+                if getkeypair_re.search(param_value):
+                    keypair = self.regxfind(getkeypair_re, param_value)
+                    param_value = 'cikey'
```

Comment:
Sure. I'll add a config file called config.ini to save these default values, load it once at the start of taskcat and use it wherever needed.